### PR TITLE
Only valid reports count for min/max batch size

### LINKED ISSUE #456: min_batch_size
> The query configuration specifies the minimum batch size, `min_batch_size`, and
maximum batch size, `max_batch_size`. The Aggregator checks that `len(X) >=
min_batch_size` and `len(X) <= max_batch_size`, where `X` is the set of reports
in the batch.

Shouldn't this be *valid* reports in the batch?

### PR DESCRIPTION
**Pull request by tgeoghegan**
Resolves #456

### CODE REVIEW

File: draft-ietf-ppm-dap.md [hunk 1]
```diff
@@ -2236,7 +2236,7 @@ shares recovered during the aggregation sub-protocol.
 ##### Size Check
 
 The query configuration specifies the minimum batch size, `min_batch_size`. The
-Aggregator checks that `len(X) >= min_batch_size`, where `X` is the set of
+Aggregator checks that `len(X) >= min_batch_size`, where `X` is the set of valid
```
Comment (tgeoghegan) [inline draft-ietf-ppm-dap.md:2239]: I considered throwing in a reference to either `{{aggregate-flow}}` or `{{validating-inputs}}` to clarify just what we mean by "valid", but I think it's clear enough: the document overall discusses report validity all over the place.

Comment (divergentdave) [reply draft-ietf-ppm-dap.md:2239]: We should do the same under `{#fixed-size-batch-validation}` as well below.

**Review by divergentdave — APPROVED** [review]

**Review by branlwyd — COMMENTED** [review]
I think saying "valid" isn't quite specific enough -- even if a given report is "valid", that doesn't necessarily mean it has completed the aggregation process. (One might argue that we must complete the aggregation process to determine if a report is valid, but several of the initial checks pre-aggregation determine "validity" of the report share.)

To avoid any wiggle room here, I think these checks should specifically check the number of reports which contributed to the values in the `CollectionReq`/`AggregateShareReq`, i.e. the number of reports in the batch which successfully completed aggregation (I believe).

**Comment by tgeoghegan** [comment]
> I think saying "valid" isn't quite specific enough -- even if a given report is "valid", that doesn't necessarily mean it has completed the aggregation process. (One might argue that we must complete the aggregation process to determine if a report is valid, but several of the initial checks pre-aggregation determine "validity" of the report share.)
> 
> To avoid any wiggle room here, I think these checks should specifically check the number of reports which contributed to the values in the `CollectionReq`/`AggregateShareReq`, i.e. the number of reports in the batch which successfully completed aggregation (I believe).

Fair enough. I changed the language from "valid reports" to "successfully aggregated reports".

**Review by branlwyd — APPROVED** [review]

**Review by cjpatton — APPROVED** [review]

**Pull request merged**